### DISCUSSION TITLE
Proper Crontab

### DISCUSSION THREAD

**Discussion by electricpollution** [Serene-Arc/bulk-downloader-for-reddit#685] [Q&A]
Hello just looking for the proper crontab syntax as what I have isn’t running. It runs by using the exact command in the terminal manually (minus spaces in between the hyphens)

current command

`0 */2 * * * python3 -m bdfr download /mnt/reddit/ - -opts /home/user/bdfr.yaml - -time-format %m_%d_%Y`

**Comment by Serene-Arc** [comment]
You'd be better served by going to a cron-specific place, especially since you haven't included any information about how it fails, but at a glance:

- You'll be running the BDFR as root; is it installed as root?
- There's spaces between the double hyphens that you have
- You may need to put the time format string in quotes

**Reply by Joeweav** [reply]
Some other thoughts.  One is whenever I do anything in cron I use full paths to everything, so make a full path to python so you know exactly what you are running. 
The bigger issue though is to add some program locking around the process so cron can not run it more than once at a time.  If for example it gets stuck, and it will, you do not want it starting up before the first go round finishes.  This is pretty simple, just check to see if a file in like /tmp exists before it starts, if the file exists, do not run the program, and if you want to be fancy, send you an email to check out what is up with it.  On the flip side, if the file does not exist, touch it so it does, run the program, and delete the file when the program is over.  The only other bit is on system start up, if the file exists, delete it.
